Content.Server/Radio/EntitySystems/JammerSystem.cs

```diff
@@ -78,11 +84,86 @@ private void OnRadioSendAttempt(ref RadioSendAttemptEvent args)
         var query = EntityQueryEnumerator<ActiveRadioJammerComponent, RadioJammerComponent, TransformComponent>();
         while (query.MoveNext(out _, out _, out var jam, out var transform))
         {
-            if (source.InRange(EntityManager, _transform, transform.Coordinates, jam.Range))
+            if (source.InRange(EntityManager, _transform, transform.Coordinates, GetCurrentRange(jam)))
             {
                 args.Cancelled = true;
                 return;
             }
         }
     }
+
+    private void OnGetVerb(Entity<RadioJammerComponent> entity, ref GetVerbsEvent<Verb> args)
+    {
+        if (!args.CanAccess || !args.CanInteract)
+            return;
+
+        var user = args.User;
+        var highVerb = new Verb
+        {
+            Priority = 3,
+            Category = VerbCategory.PowerLevel,
+            Disabled = entity.Comp.SelectedPowerLevel == 2,
+            Act = () =>
+            {
+                entity.Comp.SelectedPowerLevel = 2;
+                _popup.PopupEntity(Loc.GetString("radio-jammer-component-set-high"), user, user);
+            },
+            Text = Loc.GetString("radio-jammer-component-high"),
+        };
+
+        var mediumVerb = new Verb
+        {
+            Priority = 2,
+            Category = VerbCategory.PowerLevel,
+            Disabled = entity.Comp.SelectedPowerLevel == 1,
+            Act = () =>
+            {
+                entity.Comp.SelectedPowerLevel = 1;
+                _popup.PopupEntity(Loc.GetString("radio-jammer-component-set-medium"), user, user);
+            },
+            Text = Loc.GetString("radio-jammer-component-medium"),
+        };
+
+        var lowVerb = new Verb
+        {
+            Priority = 1,
+            Category = VerbCategory.PowerLevel,
+            Disabled = entity.Comp.SelectedPowerLevel == 0,
+            Act = () =>
+            {
+                entity.Comp.SelectedPowerLevel = 0;
+                _popup.PopupEntity(Loc.GetString("radio-jammer-component-set-low"), user, user);
+            },
+            Text = Loc.GetString("radio-jammer-component-low"),
+        };
+
+        args.Verbs.Add(highVerb);
+        args.Verbs.Add(mediumVerb);
+        args.Verbs.Add(lowVerb);
```

Comment:
Repetition can probably be removed

Reply:
I made it a for loop instead. It isn't much better because of the hard set strings but its slightly cleaner! Let me know if that's what your looking for.